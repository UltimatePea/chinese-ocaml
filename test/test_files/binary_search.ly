「：骆言测试程序：二分查找算法：」

夫「二分查找」者受 列表 目标 起始 结束 焉算法乃
  若 起始 >= 结束 则
    答 -1
  余者
    设「中点」为(起始 + 结束) / 2
    设「中值」为索引 中点 列表
    观「中值」之性
    若 中值 == 目标 则 答 中点
    若 中值 > 目标 则 答 二分查找 列表 目标 起始 中点
    余者则 答 二分查找 列表 目标 (中点 + 1) 结束
    观毕
是谓

夫「查找元素」者受 列表 目标 焉算法乃
  设「长度」为长度 列表
  答 二分查找 列表 目标 0 长度
是谓

设「有序数组」为(列开始 1 其一 3 其二 5 其三 7 其一 9 其二 11 其三 13 其一 15 其二 17 其三 19 其一 列结束)

设「结果1」为查找元素 有序数组 7
设「结果2」为查找元素 有序数组 15
设「结果3」为查找元素 有序数组 1
设「结果4」为查找元素 有序数组 8

打印 『查找7: 』
打印 结果1
打印 『查找15: 』
打印 结果2
打印 『查找1: 』
打印 结果3
打印 『查找8: 』
打印 结果4