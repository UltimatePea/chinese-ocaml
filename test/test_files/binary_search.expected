查找7: 
3
查找15: 
7
查找1: 
0
查找8: 
-1